src/hypertable.c

```diff
@@ -1187,6 +1188,31 @@ ts_hypertable_insert_blocker_trigger_add(PG_FUNCTION_ARGS)
 	PG_RETURN_OID(insert_blocker_trigger_add(relid));
 }
 
+static Datum
+create_hypertable_datum(FunctionCallInfo fcinfo, Hypertable *ht)
+{
+	TupleDesc	tupdesc;
+	Datum		values[Natts_create_hypertable];
+	bool		nulls[Natts_create_hypertable] = {false};
+	HeapTuple	tuple;
+
+	if (get_call_result_type(fcinfo, NULL, &tupdesc) != TYPEFUNC_COMPOSITE)
+		ereport(ERROR,
+				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+				 errmsg("function returning record called in "
+						"context that cannot accept type record")));
+
+	tupdesc = BlessTupleDesc(tupdesc);
+	values[AttrNumberGetAttrOffset(Anum_hypertable_id)] = Int32GetDatum(ht->fd.id);
```

Comment:
Hmm, I am a bit mixed about using the attribute numbers of `_timescaledb_internal.hypertable` as this is actually not a tuple for that table. I guess it might be OK if we make it a convention somehow that we return an "initial subset" of the table's tuple... Otherwise I think we should redefine these for this particular case to make it clear it is a distinct tuple type.

I am worried about the case, for instance, when we change one tuple and the other one breaks in unforeseen ways.

Reply:
I was mixed aswell when I added it, it works for this case since we return the first 3 columns but i will make own enum for create_hypertable, so we dont trip over this in the future.